src/test/java/fi/otavanopisto/kuntaapi/test/AbstractIntegrationTest.java

```diff
@@ -46,20 +47,22 @@ protected void waitApiListCount(String path, int count) throws InterruptedExcept
     while (true) {
       counter++;
       Thread.sleep(1000);
-      
-      int listCount = countApiList(path);
-      if (listCount == count) {
-        return;
+      try {
+        int listCount = countApiList(path);
+        if (listCount == count) {
+          return;
+        }
+        
+        if (System.currentTimeMillis() > timeout) {
+          fail(String.format("Timeout waiting for %s to have count %d", path, count));
+        }
+        
+        if ((counter % 30) == 0) {
+          logger.info(String.format("... still waiting %d items, current count %d", count, listCount));
+        }
+      } catch (JsonPathException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00108)